src/pages/instances/forms/ConvertToInstanceForm.tsx

```diff
@@ -0,0 +1,179 @@
+  close,
+  formik,
+  isLoading,
+  disableSubmit,
+  project,
+  uploadState,
+}) => {
+  const handleFileChange = async (e: ChangeEvent<HTMLInputElement>) => {
+    const { onChange } = formik.getFieldProps("imageFile");
+    onChange(e);
+
+    if (e.currentTarget.files) {
+      const file = e.currentTarget.files[0];
+      const suffix = "-instance";
+      const fileExtension = getFileExtension(file.name);
+      const sanitisedFileName = sanitizeInstanceName(
+        file.name.split(fileExtension)[0],
+      );
+      const instanceName = truncateInstanceName(sanitisedFileName, suffix);
+      await formik.setFieldValue("imageFile", file);
+      await formik.setFieldValue("name", instanceName);
+
+      // validate instance name
+      await formik.validateField("name");
+      void formik.setFieldTouched("name", true, true);
+      if (!formik.errors.name) {
+        formik.setFieldError("name", undefined);
+      }
+    }
+  };
+
+  const handleCloseModal = useCallback(() => {
+    formik.resetForm();
+    close();
+  }, [formik.resetForm, close]);
+
+  return (
+    <Modal
+      close={handleCloseModal}
+      className="convert-to-instance-modal"
+      title={
+        uploadState ? "Uploading image to LXD" : "Upload virtual machine image"
+      }
+      buttonRow={
+        <>
+          <Button
+            appearance="base"
+            className="u-no-margin--bottom"
+            type="button"
+            onClick={handleCloseModal}
+          >
+            Cancel
+          </Button>
+          <ActionButton
+            appearance="positive"
+            className="u-no-margin--bottom"
+            loading={isLoading}
+            disabled={disableSubmit}
+            onClick={() => void formik.submitForm()}
+          >
+            Upload and create
+          </ActionButton>
+        </>
+      }
+    >
+      <Form
+        onSubmit={formik.handleSubmit}
+        className={uploadState ? "u-hide" : ""}
+      >
+        <Input
+          id="image-file"
+          name="imageFile"
+          type="file"
+          label="Image file"
+          onChange={(e) => void handleFileChange(e)}
+        />
+        <Input
+          {...formik.getFieldProps("name")}
+          id="name"
+          type="text"
+          label="New instance name"
+          placeholder="Enter name"
+          error={formik.touched.name ? formik.errors.name : null}
+          disabled={!formik.values.imageFile}
+        />
+        <StoragePoolSelector
+          project={project}
+          value={formik.values.pool}
+          setValue={(value) => void formik.setFieldValue("pool", value)}
+          selectProps={{
+            id: "pool",
+            label: "Root storage pool",
+            disabled: !project || !formik.values.imageFile,
+          }}
+        />
+        <Select
+          {...formik.getFieldProps("architecture")}
+          id="architecture"
+          label="Image architecture"
+          options={supportedVMArchOptions}
```

Comment:
Can this be an architecture that the host does not support? We have some logic in the base image selector to figure out which architectures are supported by the LXD server. We might want to reuse it here.